packages/wonder-blocks-date-picker/src/util/temporal-locale-utils.ts

```diff
@@ -0,0 +1,395 @@
+import {Temporal} from "temporal-polyfill";
+import {CustomModifiers} from "./types";
+
+/**
+ * Utility functions for working with Temporal dates in react-day-picker.
+ * These replace the MomentLocaleUtils that were previously used.
+ *
+ * Question: Should we move this to a separate package?
+ */
+
+/**
+ * Format a Temporal.PlainDate using a format string.
+ * Supports a subset of moment.js format tokens for compatibility.
+ */
+export function formatDate(
+    date: Temporal.PlainDate,
+    format: string | Array<string> | null | undefined,
+    locale: string = "en",
+): string {
+    // If format is an array, use the first one
+    const formatString = Array.isArray(format) ? format[0] : format;
+
+    if (!formatString) {
+        // Default format: ISO 8601 (YYYY-MM-DD)
+        return date.toString();
```

Comment:
I see Temporal.PlainDate supports [toLocaleString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal/PlainDate/toLocaleString), so that it'll take into account the locale! Should we use that? 

 It doesn't have full browser support, though we have the temporal polyfill so maybe that functionality is covered! What do you think?